packages/sil-voting-app/src/store/proposals/actions.ts

```diff
@@ -1,64 +1,98 @@
-        creationDate: new Date('April 3, 2017 08:00:00'),
-        expiryDate: new Date('May 5, 2023 03:00:00'),
-        quorum: 20,
-        threshold: 14,
-        result: { yes: 4, no: 10, abstain: 3 },
-        vote: 'Yes',
-        status: 'Submitted',
-      },
-    },
+import {
+  getDummyChainId,
+  getDummyElectionRules,
+  getDummyElectorates,
+  getDummyProposals,
+  getDummyVote,
+} from './dummyData';
+import { AddProposalsActionType, ProposalsState } from './reducer';
+
+export async function getProposals(): Promise<ProposalsState> {
+  const electorates = getDummyElectorates();
+  const electionRules = getDummyElectionRules();
+  const proposals = getDummyProposals();
+
+  const getChainId = (proposal: Proposal): ChainId => getDummyChainId(proposal);
+
+  const getNumElectorates = (proposal: Proposal): number => {
+    const electorate = electorates.find(electorate => electorate.id === proposal.electorate.id);
+
+    let numElectorates = 0;
+    if (electorate) {
+      numElectorates = Object.keys(electorate.electors).length;
+    }
+
+    return numElectorates;
+  };
+
+  const getQuorum = (proposal: Proposal): number => {
+    const numElectorates = getNumElectorates(proposal);
+    const electionRule = electionRules.find(electionRule => electionRule.id === proposal.electionRule.id);
+
+    let quorum = 0;
+    if (electionRule && electionRule.quorum) {
+      quorum = (numElectorates * electionRule.quorum.numerator) / electionRule.quorum.denominator;
+    }
 
-    zxcvzxcv: {
-      chainId: 'zxcvzxcv',
+    return quorum;
+  };
+
+  const getThreshold = (proposal: Proposal): number => {
+    const numElectorates = getNumElectorates(proposal);
+    const electionRule = electionRules.find(electionRule => electionRule.id === proposal.electionRule.id);
+
+    let threshold = 0;
+    if (electionRule) {
+      threshold = (numElectorates * electionRule.threshold.numerator) / electionRule.threshold.denominator;
+    }
+
+    return threshold;
+  };
+
+  const getVote = (proposal: Proposal): 'Invalid' | 'Yes' | 'No' | 'Abstain' => getDummyVote(proposal);
+
+  const getStatus = (proposal: Proposal): 'Active' | 'Submitted' | 'Ended' => {
+    if (proposal.status === ProposalStatus.Submitted) {
+      if (getVote(proposal) === 'Invalid') {
+        return 'Active';
+      }
+
+      return 'Submitted';
+    }
+
+    return 'Ended';
+  };
+
+  const proposalsState = proposals.map(proposal => {
+    return {
+      chainId: getChainId(proposal),
```

Comment:
I think you can remove this chain ID, since you will only connect to one chain (the BNS chain)